### PR TITLE
chore(build): add basic makefile to make life simpler

### PR DESCRIPTION
**Pull request by poerhiza**
Add basic make file to enable users to make the binaries without docker pains...

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+GCFLAGS=-gcflags=all=-trimpath=$(GOPATH)
+ASMFLAGS=-asmflags=all=-trimpath=$(GOPATH)
+
+GOFILES=`go list ./...`
+GOFILESNOTEST=`go list ./... | grep -v test`
+
+# Make Directory to store executables
+$(shell mkdir -p ${DIR})
+
+all: freebsd linux windows
```
Comment (jpillora) [inline Makefile:16]: im okay with having commands for individual OSs

though go-releaser is used to build all https://github.com/jpillora/chisel/blob/master/.github/goreleaser.yml#L1-L3 maybe you could make a `release` command, which executes the `goreleaser.yml` file locally?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	@go get -u all
+	@go mod tidy
+
+lint: ## Lint the files
+	@go fmt ${GOFILES}
+	@go vet ${GOFILESNOTEST}
+
+test: ## Run unit tests
+	CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go test -coverprofile=${DIR}/coverage.out -race -short ${GOFILESNOTEST}
+	CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go tool cover -html=${DIR}/coverage.out -o ${DIR}/coverage.html
```
Comment (jpillora) [inline Makefile:42]: will this will fail under mac/windows?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	gocover-cobertura < ${DIR}/coverage.out > ${DIR}/coverage.xml
```
Comment (jpillora) [inline Makefile:43]: xml is for junit parsing?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,48 @@
+VERSION=$(shell git describe --abbrev=0 --tags)
+BUILD=$(shell git rev-parse HEAD)
+DIRBASE=./build
+DIR=${DIRBASE}/${VERSION}/${BUILD}/bin
+
+LDFLAGS=-ldflags "-s -w ${XBUILD} -buildid=${BUILD} -X github.com/jpillora/chisel/share.BuildVersion=${VERSION}"
+GCFLAGS=-gcflags=all=-trimpath=$(GOPATH)
```
Comment (jpillora) [inline Makefile:7]: i dont think this needs to be a GCFLAG anymore, `-trimpath` is now built into `go build`

File: Makefile (hunk 2 above)
Comment (poerhiza) [reply Makefile:42]: sadly - fixed...good catch!

File: Makefile (hunk 3 above)
Comment (poerhiza) [reply Makefile:43]: Indeed - remove?

File: Makefile (hunk 1 above)
Comment (poerhiza) [reply Makefile:16]: oh cool - updated!

File: Makefile (hunk 4 above)
Comment (poerhiza) [reply Makefile:7]: wow - that's awesome to know - updated!

**Comment by poerhiza** [comment]
@jpillora - any issues with the merge?

**Comment by jpillora** [comment]
Thanks for making the changes, looks good! Merged :)

**Comment by poerhiza** [comment]
You bet - thanks for merging!

**Pull request merged**